lib/client.rb

```diff
@@ -28,11 +99,30 @@ def get_assignment(
     )
       logger = Logger.new($stdout)
       logger.level = log_level
+      assigned_variation = get_assignment_variation(subject_key, flag_key, subject_attributes, nil, logger)
+      assigned_variation&.value
+    end
+    deprecate :get_assignment, 'the get_<typed>_assignment methods', 2024, 1
```

Comment:
Example output in stdout when this deprecated function is called

```
.NOTE: EppoClient::Client#get_assignment is deprecated; use the get_<typed>_assignment methods instead. It will be removed on or after 2024-01.
EppoClient::Client#get_assignment called from /Users/sameeran/ruby-sdk/spec/client_spec.rb:264.
```